src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileMetadataService.scala

```diff
@@ -70,7 +68,9 @@ class FileMetadataService(
       fileIds: Set[UUID] = existingFileRows.toFileTypeIds
       _ <- fileMetadataRepository.deleteFileMetadata(fileIds, distinctPropertyNames)
       addedRows <- fileMetadataRepository.addFileMetadata(generateFileMetadataRows(fileIds, distinctMetadataProperties, userId))
-      _ <- validateFileMetadataService.validateAdditionalMetadata(uniqueFileIds.toSet, input.consignmentId, distinctPropertyNames)
+      fileStatuses <- validateFileMetadataService.validateAdditionalMetadata(uniqueFileIds.toSet, input.consignmentId, distinctPropertyNames)
+      _ <- consignmentStatusService.updateMetadataConsignmentStatus(consignmentId, fileStatuses, DescriptiveMetadata)
```

Comment:
Could this update be done in a single call, with the `updateMetadataConsignmentStatus` function taking a list of types?

Reply:
Yeah, that's a good point. I've changed it to take a list of status types and removed the second call.